service/src/test/java/bio/terra/pipelines/common/utils/PipelineVariableTypesEnumTest.java

```diff
@@ -0,0 +1,272 @@
+package bio.terra.pipelines.common.utils;
+
+import static bio.terra.pipelines.common.utils.PipelineVariableTypesEnum.FILE;
+import static bio.terra.pipelines.common.utils.PipelineVariableTypesEnum.FILE_ARRAY;
+import static bio.terra.pipelines.common.utils.PipelineVariableTypesEnum.INTEGER;
+import static bio.terra.pipelines.common.utils.PipelineVariableTypesEnum.STRING;
+import static bio.terra.pipelines.common.utils.PipelineVariableTypesEnum.STRING_ARRAY;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import bio.terra.pipelines.db.entities.PipelineInputDefinition;
+import bio.terra.pipelines.testutils.BaseTest;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class PipelineVariableTypesEnumTest extends BaseTest {
+
+  private static Stream<Arguments> castValidations() {
+    String commonInputName = "inputName";
+    // error messages
+    String stringTypeErrorMessage = "%s must be a string".formatted(commonInputName);
+    String integerTypeErrorMessage = "%s must be an integer".formatted(commonInputName);
+    String vcfTypeErrorMessage =
+        "%s must be a path to a file ending in .vcf.gz".formatted(commonInputName);
+    String stringArrayTypeErrorMessage =
+        "%s must be an array of strings".formatted(commonInputName);
+    String vcfArrayTypeErrorMessage =
+        "%s must be an array of paths to files ending in .vcf.gz".formatted(commonInputName);
+    String notNullOrEmptyErrorMessage = "%s must not be null or empty".formatted(commonInputName);
+
+    // the only used information in the input definitions is name, type, and fileSuffix
+    PipelineInputDefinition integerInputDefinition =
+        new PipelineInputDefinition(
+            1L, commonInputName, "integer_input", INTEGER, null, true, true, null);
+    PipelineInputDefinition stringInputDefinition =
+        new PipelineInputDefinition(
+            1L, commonInputName, "string_input", STRING, null, true, true, null);
+    PipelineInputDefinition fileVcfInputDefinition =
+        new PipelineInputDefinition(
+            1L, commonInputName, "file_vcf_input", FILE, ".vcf.gz", true, true, null);
+    PipelineInputDefinition fileBedInputDefinition =
+        new PipelineInputDefinition(
+            1L, commonInputName, "file_bed_input", FILE, ".bed", true, true, null);
+    PipelineInputDefinition fileNoSuffixInputDefinition =
```

Comment:
@jsotobroad  here + line 116 is the test I added to show we can define a FILE input without a suffix check by using an empty string as the fileSuffix value.